am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java

```diff
@@ -0,0 +1,54 @@
+package integration.uk.gov.hmcts.reform.amlib;
+
+import com.fasterxml.jackson.core.JsonPointer;
+import integration.uk.gov.hmcts.reform.amlib.base.IntegrationBaseTest;
+import org.junit.jupiter.api.Test;
+import uk.gov.hmcts.reform.amlib.enums.Permission;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.CREATE_PERMISSION;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.READ_PERMISSION;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.RESOURCE_NAME;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.RESOURCE_TYPE;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.ROLE_NAME;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.SERVICE_NAME;
+
+class GetRolePermissionsIntegrationTest extends IntegrationBaseTest {
+
+    @Test
+    void returnListOfPermissionsForRoleName() {
+        Map<JsonPointer, Set<Permission>> accessRecord = ams.getRolePermissions(SERVICE_NAME,
+            RESOURCE_TYPE, RESOURCE_NAME, ROLE_NAME);
+
+        assertThat(accessRecord.containsKey("/test") && accessRecord.containsValue(READ_PERMISSION));
```

Comment:
I suggest adding another assertion checking size of the result map as well.

Reply:
👍